### PR TITLE
[DT-356][risk=no] Implement prepackaged sets for individual surveys

### PR DESCRIPTION
**Pull request by chenchals**
**Description**
---
- implemented selection for individual / combination of surveys or All surveys 
- updated UI - individual surveys are tabbed-in under `All Surveys`
image

- Test: added a sort for standard and source... for test to pass - need to fix test for correct SQL generation comparison.

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md)
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md)
- [x] If this PR is intended to complete a JIRA story, I have checked that all AC are met for that story
- [ ] I have manually tested this change and my testing process is described above
- [ ] This PR includes appropriate automated tests, and I have documented any behavior that cannot be tested with code
- [ ] If this fixes a bug, ensure the steps to reproduce are in the Jira ticket or provided above.
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented the impacts in the description
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this includes an API change, I have run the relevant E2E tests locally because API changes are not covered by our PR checks

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java [hunk 1]
```diff
@@ -752,12 +781,24 @@ private Optional<String> buildConceptIdSqlInClause(
...
-      if (!prePackagedSurveyConceptSet(dbConceptSets)) {
-        dbConceptSetConceptIds = findDomainConceptIds(domain, dbConceptSetIds);
+      if (prePackagedAllSurveyConceptSet(dbConceptSets)) {
+        return Optional.empty();
+      }
+      if (userSurveyConceptSet(dbConceptSets)) {
+        dbConceptSetConceptIds.addAll(findDomainConceptIds(domain, dbConceptSetIds));
       }
+      // handle prepackaged PFHH
+      if (prePackagedPfhhSurveyConceptSet(dbConceptSets)) {
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java:795]: Shouldn't this be behind the feature flag?

if (workbenchConfigProvider.get().featureFlags.enableDataExplorer)

File: api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java [hunk 2]
```diff
@@ -390,16 +414,16 @@ public TableResult previewBigQueryJobConfig(DataSetPreviewRequest request) {
                 dbConceptSetConceptIds.stream()
                     .filter(cid -> !pfhhSurveyQuestionIds.contains(cid.getConceptId()))
                     .collect(Collectors.toList());
-            dbConceptSetConceptIds = dbNonPFHHSurveyQuestions;
+            dbConceptSetConceptIds.addAll(dbNonPFHHSurveyQuestions);
```
[the remark this answers is not in the document]
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java]: Thank you. Missed this, fixed.

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 3]
```diff
@@ -1092,13 +1131,67 @@ export const DatasetPage = fp.flow(
...
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYBASICS);
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYLIFESTYLE);
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYOVERALLHEALTH);
+          updatedPrepackaged.add(
+            PrePackagedConceptSetEnum.SURVEYHEALTHCAREACCESSUTILIZATION
+          );
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYCOPE);
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYSDOH);
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYCOVIDVACCINE);
+          updatedPrepackaged.add(PrePackagedConceptSetEnum.SURVEYPFHH);
```
Comment (dolbeew) [inline ui/src/app/pages/data/data-set/dataset-page.tsx:1145]: Can iterate on `PREPACKAGED_SURVEY_DOMAINS` to make this a little cleaner:
```suggestion
          Object.keys(PREPACKAGED_SURVEY_DOMAINS).forEach((prepackagedSurvey) =>
            updatedPrepackaged.add(prepackagedSurvey)
          );
```

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 4]
```diff
@@ -1092,13 +1131,67 @@ export const DatasetPage = fp.flow(
...
+          );
+          updatedPrepackaged.delete(
+            PrePackagedConceptSetEnum.SURVEYHEALTHCAREACCESSUTILIZATION
+          );
+          updatedPrepackaged.delete(PrePackagedConceptSetEnum.SURVEYCOPE);
+          updatedPrepackaged.delete(PrePackagedConceptSetEnum.SURVEYSDOH);
+          updatedPrepackaged.delete(
+            PrePackagedConceptSetEnum.SURVEYCOVIDVACCINE
+          );
+          updatedPrepackaged.delete(PrePackagedConceptSetEnum.SURVEYPFHH);
```
Comment (dolbeew) [inline ui/src/app/pages/data/data-set/dataset-page.tsx:1170]: Same as above for the deletes here:
```suggestion
          Object.keys(PREPACKAGED_SURVEY_DOMAINS).forEach((prepackagedSurvey) =>
            updatedPrepackaged.delete(prepackagedSurvey)
          );
```

**Review by dolbeew — APPROVED** [review]
UI looks good, just a couple suggestions

File: ui/src/app/pages/data/data-set/dataset-page.tsx (hunk 3 above)
Comment (chenchals) [reply ui/src/app/pages/data/data-set/dataset-page.tsx:1145]: Thanks. I tried that... maybe I had a bug. Committing you suggestion

**Review by freemabd — APPROVED** [review]

**Pull request merged**